### PR TITLE
#12147: Add queue_id and optional output tensors to ttnn.neg_bw

### LINKED ISSUE #12147: Add queue_id and optional output tensors to ttnn.neg
Master Issue : #9123
**STATUS : PR Review** https://github.com/tenstorrent/tt-metal/pull/12388

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Issue : https://github.com/tenstorrent/tt-metal/issues/12147
https://github.com/tenstorrent/tt-metal/actions/runs/10773054531 - passed

Test File : `tests/ttnn/unit_tests/operations/backward/test_backward_neg.py`
Screenshot 2024-09-09 at 7 26 17 PM

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/backward/test_backward_neg.py [hunk 1]
```diff
@@ -27,3 +27,54 @@ def test_bw_neg(input_shapes, device):
...
+    ),
+)
+def test_bw_neg_opt(input_shapes, device):
+    in_data, input_tensor = data_gen_with_range(input_shapes, -100, 100, device, True)
+    grad_data, grad_tensor = data_gen_with_range(input_shapes, -100, 100, device)
+    _, input_grad = data_gen_with_range(input_shapes, -1, 1, device)
+    input_grad = ttnn.to_memory_config(input_grad, ttnn.L1_MEMORY_CONFIG)
+
+    pages_before = ttnn._ttnn.reports.get_buffer_pages()
+    tt_output_tensor_on_device = ttnn.neg_bw(grad_tensor, input_tensor, input_grad=input_grad)
```
Comment (Aswinmcw) [inline tests/ttnn/unit_tests/operations/backward/test_backward_neg.py:47]: Should not assign the value to left side

File: tests/ttnn/unit_tests/operations/backward/test_backward_neg.py [hunk 2]
```diff
@@ -27,3 +27,54 @@ def test_bw_neg(input_shapes, device):
...
+)
+def test_bw_neg_opt_id(input_shapes, device):
+    in_data, input_tensor = data_gen_with_range(input_shapes, -100, 100, device, True)
+    grad_data, grad_tensor = data_gen_with_range(input_shapes, -100, 100, device)
+    _, input_grad = data_gen_with_range(input_shapes, -1, 1, device)
+    input_grad = ttnn.to_memory_config(input_grad, ttnn.L1_MEMORY_CONFIG)
+    cq_id = 0
+
+    pages_before = ttnn._ttnn.reports.get_buffer_pages()
+    tt_output_tensor_on_device = ttnn.neg_bw(grad_tensor, input_tensor, input_grad=input_grad, queue_id=cq_id)
```
Comment (Aswinmcw) [inline tests/ttnn/unit_tests/operations/backward/test_backward_neg.py:73]: Same here

**Review by Aswinmcw — APPROVED** [review]
LGTM

**Review by eyonland — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/10789752332 - passed

**Pull request merged**